optuna/integration/cma.py

```diff
@@ -310,17 +310,17 @@ def __init__(
                 highs.append(len(dist.choices) - 0.5)
             elif isinstance(dist, UniformDistribution) or isinstance(dist, LogUniformDistribution):
                 lows.append(self._to_cma_params(search_space, param_name, dist.low))
-                highs.append(self._to_cma_params(search_space, param_name, dist.high))
+                highs.append(self._to_cma_params(search_space, param_name, dist.high) - _EPS)
```

Comment:
If the value of `low` and `high` is relatively the same order with `_EPS`, that is, `low ≒ high ≒ 1e-10`, then this subtraction may cause an invalid interval `[low, high)`. How about subtracting the relatively small values as follows?
```python
low = self._to_cma_params(search_space, param_name, dist.low)
high = self._to_cma_params(search_space, param_name, dist.high)
lows.append(low)
highs.append(high - (high - low) * _EPS)
```

Reply:
We can easily reproduce the representation error which is grater than the situation that you assumed.

```python
>>> high = 1.1e-10
>>> low = 0.9e-10
>>> diff = (high - low) * 1e-10
>>> representation_error = abs(0.1 - (0.3-0.2))
>>> representation_ > diff
True
```

So I think we should ignore the edge case. What do you think?

Reply:
I mean if we should consider such a situation, we should define `_EPS = 1e-11` or more tiny values.

Reply:
Thanks for the clarification! I understood what you mean. I totally agree with you and resolve this conversation without any fixes.